src/LinuxTracing/PerfEvent.h

```diff
@@ -119,6 +119,16 @@ struct UprobesPerfEventData {
 };
 using UprobesPerfEvent = TypedPerfEvent<UprobesPerfEventData>;
 
+struct UprobesWithStackPerfEventData {
+  pid_t pid;
+  pid_t tid;
+  perf_event_sample_regs_user_sp regs;
+
+  uint64_t dyn_size;
+  mutable std::unique_ptr<char[]> data;
```

Comment:
You should add a comment on the mutability like it's done in other `struct`s here with `mutable` fields.